### ISSUE TITLE
Fix imprecise start/end indices in rules using `node.selectors`

### ISSUE THREAD

**Issue by mattxwang** [stylelint/stylelint#6075]
### What is the problem you're trying to solve?

In the process of working on #5694, I've noticed a couple of edge-cases that could use more precise/improved warning indices.

For example, in `no-invalid-double-slash-comments` (related PR: #6072), I note:

> For `no-invalid-double-slash-comments`, there was one test case I was confused about:
> 
> ```js
> code: 'a, //div { color: pink; }',
> description: 'between rules',
> message: messages.rejected,
> line: 1,
> column: 1,
> endLine: 1,
> endColumn: 26,
> ```
> 
> I haven't changed the behaviour of this test case, but to me I'm not sure why it's `column: 1` - the comment starts at index `4` (where the double-slash is). I might be misunderstanding how this rule works, though!

And in `no-descending-specificity` (related PR: #6049), @ybiquitous notes:

> I'm concerned about the following edge-case:
> 
> ```js
> {
> 	code: 'a:hover {} a, b, a {}',
> 	warnings: [
> 		{
> 			message: messages.rejected('a', 'a:hover'),
> 			line: 1,
> 			column: 12,
> 			endLine: 1,
> 			endColumn: 13,
> 		},
> 		{
> 			message: messages.rejected('a', 'a:hover'),
> 			line: 1,
> 			column: 18,    // but actual: 12
> 			endLine: 1,
> 			endColumn: 19, // but actual: 13
> 		},
> 	],
> },
> ```

There might be many more of these edge-cases that report slightly-mismatched/erroneous indices for the offending warnings. More precise indices can be very helpful downstream, especially for the vscode-stylelint team.

### What solution would you like to see?

@ybiquitous has suggested to use `postcss-selector-parser` instead of `ruleNode.selectors` to provide us with exact positions for selectors instead of guessing. Depending on whether or not other issues involve other elements, we may have to use other postcss parsers as well.

Overall, I'm not very familiar with the internals, so any comments are helpful! If we agree this is a good idea, I can get some of this work done - at the very least, address the issue for the two rules mentioned above, and start working on these as I encounter more rules in #5694.

---

- [ ] `no-duplicate-selectors`
- [x] `no-invalid-double-slash-comments`
- [x] `no-descending-specificity`

**Comment by ybiquitous** [comment]
I think there is the same problem in `no-duplicate-selectors`. See #6047.

**Comment by ybiquitous** [comment]
I cannot find a PR to fix, but the `no-descending-specificity` issue has been resolved. See the [demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcICG8AWEBuMBOACYAvrkgDS4BGZS+BIJIAZgJYA2MAckgLZyIwAe3AA5sAdGADOEuuAgA7ZgHMEIYAB05uXGpDYArmwk74+DVq065EALSwJYGHKhM5i6xKEwwTZt4AuAJ7GuH76MGa4BBq09JAKTIoAYhDYXEh+KgBWEvIysELSiOqa2iASgWwsLn7BOizpMOU60SAEQA).
